### PR TITLE
EQL: implement case sensitivity for indexOf and endsWith string functions

### PR DESCRIPTION
**Pull request by aleksmaus**
Addresses: https://github.com/elastic/elasticsearch/issues/56750

Thanks @astefan for paving the way!

### CODE REVIEW

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWith.java [hunk 1]
```diff
@@ -86,14 +92,14 @@ public ScriptTemplate asScript() {
...
-                "endsWith",
-                sourceScript.template(),
-                patternScript.template()),
-                paramsBuilder()
-                    .script(sourceScript.params())
-                    .script(patternScript.params())
-                    .build(), dataType());
+        ParamsBuilder params = paramsBuilder();
+
+        String template = formatTemplate("{eql}.endsWith(" + sourceScript.template() + "," + patternScript.template() + ",{})");
```
Comment (rw-access) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWith.java:97]: We're not totally consistent in these. Here's something I did for `stringContains`
https://github.com/elastic/elasticsearch/blob/8fb5b566df3703aae22fcd75db3527e256dd44b4/x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/StringContains.java#L87-L98

**Review by rw-access — APPROVED** [review]

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWith.java (hunk 1 above)
Comment (aleksmaus) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWith.java:97]: Followed Andrei implementation for consistency, can change back. Let me know.

Comment (aleksmaus) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/EndsWith.java]: Updated PR

**Pull request merged**